src/library/scala/collection/MapView.scala

```diff
@@ -21,23 +24,35 @@ trait MapView[K, +V]
 
   override def view: MapView[K, V] = this
 
+  def concat[V1 >: V](that: SomeMapOps[K, V1]): MapView[K, V1] = mapFactory.from(new MapView.Concat(this, that))
+
+  def ++[V1 >: V](that: SomeMapOps[K, V1]): MapView[K, V1] = concat(that)
+
+  override def +[V1 >: V](kv: (K, V1)): MapView[K, V1] = concat(new Map1(kv._1, kv._2))
+
   /** Filters this map by retaining only keys satisfying a predicate.
     *  @param  p   the predicate used to test keys
     *  @return an immutable map consisting only of those key value pairs of this map where the key satisfies
     *          the predicate `p`. The resulting map wraps the original map without copying any elements.
     */
-  override def filterKeys(p: K => Boolean): MapView[K, V] = new MapView.FilterKeys(this, p)
+  override def filterKeys(p: K => Boolean): MapView[K, V] = mapFactory.from(new MapView.FilterKeys(this, p))
 
   /** Transforms this map by applying a function to every retrieved value.
     *  @param  f   the function used to transform values of this map.
     *  @return a map view which maps every key of this map
     *          to `f(this(key))`. The resulting map wraps the original map without copying any elements.
     */
-  override def mapValues[W](f: V => W): MapView[K, W] = new MapView.MapValues(this, f)
+  override def mapValues[W](f: V => W): MapView[K, W] = mapFactory.from(new MapView.MapValues(this, f))
+
+  override def filter(pred: ((K, V)) => Boolean): MapView[K, V] = mapFactory.from(new MapView.Filter(this, false, pred))
+
+  override def filterNot(pred: ((K, V)) => Boolean): MapView[K, V] = mapFactory.from(new MapView.Filter(this, true, pred))
 
-  def mapFactory: MapFactory[({ type l[X, Y] = View[(X, Y)] })#l] = new MapView.MapViewMapFactory[K, V]
+  override def partition(p: ((K, V)) => Boolean): (MapView[K, V], MapView[K, V]) = (filter(p), filterNot(p))
 
-  def empty: View[(K, V)] = View.Empty
+  def mapFactory: MapView.MapFactory = MapView.MapFactory
+
+  def empty: MapView[K, V] = mapFactory.empty
 }
 
 object MapView {
```

Comment:
The reason I didn't make `MapView` extend `MapFactory` is because then you would have this confusing syntax:

```
MapView(1 -> 2) // returns View[(Int, Int)], not a MapView[Int, Int]
```

Maybe the apply method should be changed so that it does return MapView[K, V]. It would have to of course strictly convert the vararg into a `Map[K, V]` first though, and @julienrf had wanted to avoid this in other scenarios in a previous PR.